tools/gethostlatency.py

```diff
@@ -81,13 +97,22 @@
     return 0;
 }
 """
+
+# We really mean the tgid from the kernel's perspective, which is in
+# the top 32 bits of bpf_get_current_pid_tgid().
+if args.pid == -1:
+    bpf_text = bpf_text.replace('FILTER', '')
+else:
+    bpf_text = bpf_text.replace('FILTER',
+        "if (pid != %d) { return 0; }" % args.pid)
+
 b = BPF(text=bpf_text)
-b.attach_uprobe(name="c", sym="getaddrinfo", fn_name="do_entry")
-b.attach_uprobe(name="c", sym="gethostbyname", fn_name="do_entry")
-b.attach_uprobe(name="c", sym="gethostbyname2", fn_name="do_entry")
-b.attach_uretprobe(name="c", sym="getaddrinfo", fn_name="do_return")
-b.attach_uretprobe(name="c", sym="gethostbyname", fn_name="do_return")
-b.attach_uretprobe(name="c", sym="gethostbyname2", fn_name="do_return")
+b.attach_uprobe(name="c", sym="getaddrinfo", fn_name="do_entry", pid=args.pid)
+b.attach_uprobe(name="c", sym="gethostbyname", fn_name="do_entry", pid=args.pid)
+b.attach_uprobe(name="c", sym="gethostbyname2", fn_name="do_entry", pid=args.pid)
+b.attach_uretprobe(name="c", sym="getaddrinfo", fn_name="do_return", pid=args.pid)
+b.attach_uretprobe(name="c", sym="gethostbyname", fn_name="do_return", pid=args.pid)
+b.attach_uretprobe(name="c", sym="gethostbyname2", fn_name="do_return", pid=args.pid)
```

Comment:
Is the filter in the BPF program itself really required if you're attaching the uprobe to a specific process? I think not.